### DISCUSSION TITLE
does the react-square-web-payments-sdk perform the payment transaction ?

### DISCUSSION THREAD

**Discussion by ebizcoAU** [weareseeed/react-square-web-payments-sdk#24] [Help] [answered]
I used the guided demo to do a payment using credit card with applicationId and locationId from Square Sandbox Account. 
All I got is the token response ... No transaction is taken place in my Sandbox account...
Soln: From the token response, one should proceed with "/payment" as shown in server.js in Square Web Payment SDK....
Some extra notes would greatly help..

**Comment by sfrunza** [comment, marked as answer]
I had to do more setup in Rails backend with **gem 'square.rb'** and create a post method like so 
```
  def create_payment(nonce, price)
     client = Square::Client.new(
         access_token: ENV['SQUARE_ACCESS_TOKEN'],
         environment: "sandbox"
     )
     location_id = ENV['LOCATION_ID']
     result = client.payments.create_payment(
            body: {
                source_id: nonce,
                idempotency_key: SecureRandom.uuid(),
                amount_money: {
                     amount: price,
                     currency: "USD"
                },
                location_id: location_id
            }
      )
      return result
   end

```
and from React 

```
cardTokenizeResponseReceived={(token, buyer) => {
        let price = 10000 // $100.00
        const body = JSON.stringify({
          price: price,
          nonce: token.token,
        });

        fetch("/payments", {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body,
        })
          .then((res) => res.json())
          .then((data) => {
            if (data.payment.errors) {
              console.log(data.payment.error)
            } else {
              console.log(data)
            }
          });
      }}

**Comment by danestves** [comment]
Hi @ebizcoAU the React form is just a wrapper of the Payments SDK, the Payments SDK doesn't take payments

Just generates the tokens and information needed for you as a developer or a third-party service to receive this data and process a payment using a custom backend

As @sfrunza shows he is sending the data to his backend after the response received
